components/tabs/src/storage.rs

```diff
@@ -72,15 +78,26 @@ pub struct ClientRemoteTabs {
 pub struct TabsStorage {
     local_tabs: RefCell<Option<Vec<RemoteTab>>>,
     db_path: PathBuf,
-    db_connection: Option<Connection>,
+    db_connection: DbConnection,
 }
 
 impl TabsStorage {
     pub fn new(db_path: impl AsRef<Path>) -> Self {
         Self {
             local_tabs: RefCell::default(),
             db_path: db_path.as_ref().to_path_buf(),
-            db_connection: None,
+            db_connection: DbConnection::Created,
+        }
+    }
+
+    pub fn close(&mut self) -> Result<bool> {
```

Comment:
```suggestion
    pub fn close(&mut self) -> Result<()> {
```

Minor nit: It's a bit more idiomatic to use `()` when the result isn't meaningful.

Reply:
decided to go with your below approach and remove this completely!